### PR TITLE
Provider: Mailchimp Open Commerce (formerly Reaction Commerce)

### PR DESCRIPTION
**Pull request by loan-laux**
This is a work in progress and is not ready for production yet.

**Status**

* [x]  CommerceProvider
* [x]  Schema & TS types
* [x]  API Operations - Get all collections
* [x]  API Operations - Get all pages
* [x]  API Operations - Get all products
* [ ]  API Operations - Get page
* [x]  API Operations - Get product
* [x]  API Operations - Get Shop Info (categories and vendors working — `vendors` query still a WIP PR on Reaction)
* [x]  Hook - Add Item
* [x]  Hook - Remove Item
* [x]  Hook - Update Item
* [x]  Hook - Get Cart (account-tied carts working, anonymous carts working, cart reconciliation working)
* [x]  Auth (based on a WIP PR on Reaction - still need to implement refresh tokens)
* [x]  Customer information
* [x]  Product attributes - Size, Colors
* [ ]  Custom checkout
* [ ]  Typing (in progress)
* [ ]  Tests

@patrys, I hope you don't mind me stealing your great status list here.

The code is far from what I'd consider clean so don't judge me. I used the Shopify provider as a base, so some bits of it might still be there and I'll clean them up at a later point. The use of types is also questionable currently as it's my first time actually using TS, but I would say the priority for now is to get something that works.

At this stage, I'm creating this PR so that people can discuss changes and how to best approach this integration.

### CODE REVIEW

**Comment by aldeed** [comment]
Nice @loan-laux ! I was just debating whether to try creating this sometime and then spotted your PR. I don't have too much time to help, but I can review and test when it's done if you want. Or let me know if you get stuck on something.

File: package.json [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
   "scripts": {
-    "dev": "next dev",
+    "dev": "next dev -p 8080",
     "build": "next build",
     "start": "next start",
     "analyze": "BUNDLE_ANALYZE=both yarn build",
     "prettier-fix": "prettier --write .",
     "find:unused": "next-unused",
     "generate": "graphql-codegen",
+    "generate:reactioncommerce": "graphql-codegen --config framework/reactioncommerce/codegen.json",
```
Comment (okbel) [inline package.json:12]: @lfades  I wonder if it's better to add a .graphqlrc, or .graphqlrc.yml/yaml instead. wdyt?
Also, the buildsteps can be in the Vercel project.

Comment (okbel) [reply package.json:12]: Discussed with @lfades. We're good for now 👍

**Comment by lfades** [comment]
@loan-laux Feel free to ping me when the PR is ready for a review 👍

**Comment by goncy** [comment]
Hi @loan-laux, is this still being worked on?

**Comment by loan-laux** [comment]
Hey @goncy, I've been pretty busy lately but planning to get this through the finish line. Just need to get it back in sync with the main branch and finish the checkout integration.

**Comment by goncy** [comment]
> Hey @goncy, I've been pretty busy lately but planning to get this through the finish line. Just need to get it back in sync with the main branch and finish the checkout integration.

Great then, I will leave it open, thanks!

**Comment by leerob** [comment]
Hey there! Thank you for your contribution. We have decided to take Next.js Commerce in a new direction and will be closing out current PRs and issues due to this change. Please see this PR for more details: https://github.com/vercel/commerce/pull/966

Would love to see an Open Commerce version of v2!

**Pull request closed without merging**